images/egress/dns-proxy/egress-dns-proxy.sh

```diff
@@ -80,12 +80,35 @@ function generate_dns_resolvers() {
     echo ""
 }
 
+function get_num_servers() {
+    local domain=$1
+
+    if [[ "${EGRESS_DNS_PROXY_MODE:-}" == "unit-test" ]]; then
+        if [[ "${DNS_SERVERS[${domain}]+isset}" ]]; then
+            echo "${DNS_SERVERS[${domain}]}"
+        else
+            echo "1"
+        fi
+        return
+    fi
+
+    declare -A servers=()
+    for ns in ${NAMESERVERS[@]}; do
+        local output=$( dig +nocmd +noall +answer a "${domain}" @"${ns}" | awk '{ print $5 }' )
+        for ip in $( echo $output | xargs ); do
+            servers[${ip}]=1
+        done
+    done
+    echo "${#servers[@]}"
```

Comment:
Seems like you could replace the `for` loop and this `echo` command with `ips=( $output ); echo ${#ips[@]}`.  Are you guarding against duplicate A records?

Reply:
Yes, multiple dns nameservers could return the same A record.
ips=( $output ) may not remove the duplicates.